src/pages/blog/starlingx-single-core-platform.md

```diff
@@ -0,0 +1,60 @@
+
+### Platform Core Adjustments
+
+## System Inventory
+
+# Periodic and Runtime Tasks
+
+Some of the sysinv-conductor and sysinv-agent periodic tasks have been redesigned. The main changes focused on refactoring legacy code, increasing time intervals, and converting periodic tasks to on-demand tasks, whenever possible.
+
+# Remote Procedure Calls
+
+In previous versions, RPC communication between sysinv-api, sysinv-conductor and sysinv-agent processes are performed using an openstack RPC library, with RabbitMQ as the communication transport layer. To decouple sysinv from RabbitMQ, using a brokerless solution instead, a new backend was developed. This backend was implemented using a library built on top of ZeroMQ for message queuing and MessagePack for data serialization.
+
+## Scale-Down Services
+
+Platform services are typically designed with threads or worker processes that are proportional to the number of available cores on the platform. However, simply reducing the number of cores to just one may not be sufficient, as each service also has a minimum scale requirement. To find a balance between minimum scale and overall performance, a scale-down approach was implemented for certain services.
+
+This scale-down process involves reducing the number of threads and workers assigned to services such as postgres, etcd, containerd, memcached, armada, keystone, barbican, docker-registry, docker-token-server, kube-apiserver, and kubelet. It is important to note that worker allocation rules remain the same, with the changes affecting only applications that run on single core.
+
+Numerous studies have demonstrated that optimal efficiency is achieved when the number of threads aligns with the available cores. This approach also helps to prevent any process from monopolizing the CPU, promoting fair resource allocation across the system.
+
+## Postgres Tuning
+
+Adjustments have been made to certain parameters of Postgres services. These modifications aim to ensure a harmonious balance between system demand and CPU usage. Specifically, changes have been implemented in the number of workers, autovacuum operators, and the frequency of autovacuum operations. By fine-tuning these aspects, the system can achieve optimal performance while efficiently managing database operations.
+
+## Other optimizations
+
+As part of our platform optimization efforts, we investigated services that consumed significant computational resources. This investigation led to merging related services to reduce overall overhead and removing services that were no longer necessary.
```

Comment:
```suggestion
As part of the platform optimization efforts, an investigation was conducted to identify services that consumed significant computational resources. As a result, the overall system overhead was reduced by  merging related services and removing components that were no longer necessary.
```

Reply:
Do you have examples of what was merged or removed?

Reply:
fpga-agent merged with sysinv-agent, and sm-watchdog was removed. But I'm not sure we should mention that.